btf_interp/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! Provide interpreter implementation for BF program.
+use btf_types::BrainFuckProgram;
+
+/// Provide structure for Virtual Machine
+#[derive(Debug)]
+pub struct VirtualMachine<T> {
+    /// The collection to store elements of the tape.
+    tape: Vec<T>,
+    /// The size of the tape.
+    tape_size: usize,
+    /// Whether to allow adjust size of the tape of not.
+    adjust_tape: bool,
+    /// The pointer to the current element of tape.
+    pointer: usize,
+}
+
+impl<T> VirtualMachine<T> {
+    /// Create VM based on the size, by default is 30 000. Also, it can be adjusted, by default it doesn't.
+    pub fn new(size: Option<usize>, adjust_tape: Option<bool>) -> Self {
+        VirtualMachine {
+            tape: Vec::new(),
+            tape_size: size.unwrap_or(30000),
```

Comment:
This is good, but what would happen if we passed in `Some(0)` ?

A good idea can be to make invalid types unrepresentable, perhaps consider looking at the `NonZeroUsize` type for the argument to `new()`